examples/step-55/step-55.cc

```diff
@@ -203,19 +203,22 @@ namespace Step55
     const double R_x = p[0];
     const double R_y = p[1];
 
-    const double pi  = numbers::PI;
-    const double pi2 = pi * pi;
-    values[0] =
-      -1.0L / 2.0L * (-2 * sqrt(25.0 + 4 * pi2) + 10.0) *
-        exp(R_x * (-2 * sqrt(25.0 + 4 * pi2) + 10.0)) -
-      0.4 * pi2 * exp(R_x * (-sqrt(25.0 + 4 * pi2) + 5.0)) * cos(2 * R_y * pi) +
-      0.1 * pow(-sqrt(25.0 + 4 * pi2) + 5.0, 2) *
-        exp(R_x * (-sqrt(25.0 + 4 * pi2) + 5.0)) * cos(2 * R_y * pi);
-    values[1] = 0.2 * pi * (-sqrt(25.0 + 4 * pi2) + 5.0) *
-                  exp(R_x * (-sqrt(25.0 + 4 * pi2) + 5.0)) * sin(2 * R_y * pi) -
-                0.05 * pow(-sqrt(25.0 + 4 * pi2) + 5.0, 3) *
-                  exp(R_x * (-sqrt(25.0 + 4 * pi2) + 5.0)) * sin(2 * R_y * pi) /
-                  pi;
+    constexpr double pi  = numbers::PI;
```

Comment:
Out of curiosity: What is the practical difference between ``const`` and ``constexpr`` here? That it could be evaluated at compile time? Would it, though? I would assume the optimizer gets rid of this variable anyways...

Reply:
Yes, the compiler *must* evaluate the initializer at compile-time. For `pi` this makes no difference, of course: It's just copying one `constexpr` variable to another. For the `pi2` variable, it will actually have to do a bit of work.

One might surmise that a good compiler will do the same without `constexpr` (and also without `const`), but it probably does not hurt.